### PR TITLE
Migrate notifcation job to work manager api

### PR DESCRIPTION
**Pull request by ezaquarii**

### CODE REVIEW

**Comment by tobiasKaminsky** [comment]
@ezaquarii this has some heavy dagger problems:

```
gplayDebug/com/nextcloud/client/jobs/NotificationWork.java:78: error: Dagger does not support injection into private fields
        private com.nextcloud.client.account.UserAccountManager accountManager;
```

**Comment by ezaquarii** [comment]
On Tue, Apr 28, 2020 at 11:28:11AM -0700, Tobias Kaminsky wrote:
> @ezaquarii this has some heavy dagger problems:

@tobiasKaminsky Yeah, there is a bug in Dagger - field injection into
nested Kotlin classes doesn't work.

For some reason I dropped a fix commit for this problem
when doing a rebase.

**Comment by tobiasKaminsky** [comment]
I tested it and sent me a notification, but got:
```
2020-04-29 14:04:15.724 28337-28384/com.nextcloud.client E/WM-WorkerFactory: Could not instantiate com.nextcloud.client.jobs.NotificationWork
    java.lang.NoSuchMethodException: com.nextcloud.client.jobs.NotificationWork.<init> [class android.content.Context, class androidx.work.WorkerParameters]
        at java.lang.Class.getConstructor0(Class.java:2332)
        at java.lang.Class.getDeclaredConstructor(Class.java:2170)
        at androidx.work.WorkerFactory.createWorkerWithDefaultFallback(WorkerFactory.java:95)
        at androidx.work.impl.WorkerWrapper.runWorker(WorkerWrapper.java:242)
        at androidx.work.impl.WorkerWrapper.run(WorkerWrapper.java:136)
        at androidx.work.impl.utils.SerialExecutor$Task.run(SerialExecutor.java:91)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
2020-04-29 14:04:15.724 28337-28384/com.nextcloud.client E/WM-WorkerWrapper: Could not create Worker com.nextcloud.client.jobs.NotificationWork
```

**Comment by tobiasKaminsky** [comment]
Fixed, tested and worked :+1:

File: src/main/java/com/nextcloud/client/jobs/BackgroundJobFactory.kt [hunk 1]
```diff
@@ -160,4 +163,13 @@ class BackgroundJobFactory @Inject constructor(
...
         )
     }
+
+    private fun createNotificationWork(context: Context, params: WorkerParameters): NotificationWork {
+        return NotificationWork(
+            context,
+            params,
+            notificationManager,
+            accountManager
+                               )
```
Comment (ezaquarii) [inline src/main/java/com/nextcloud/client/jobs/BackgroundJobFactory.kt:173]: I noticed that Android Studio started doing this wonky formatting recently. So far I was just fixing it by hand, but maybe we should think about digging into the projects' style config.

Comment (tobiasKaminsky) [reply src/main/java/com/nextcloud/client/jobs/BackgroundJobFactory.kt:173]: I opened a new issue for this, so we can merge this PR.

**Pull request merged**